modules/core/src/lib/deck.js

```diff
@@ -275,6 +275,8 @@ export default class Deck {
     // If initialized, update sub manager props
     if (this.layerManager) {
       this.viewManager.setProps(resolvedProps);
+      // Make sure that any new layer gets initialized with the current viewport
+      this.layerManager.activateViewport(this.getViewports()[0]);
```

Comment:
Is there any chance that there are no views? I seem to recall an early iteration of layer manager that maintained a dummy view...

Reply:
Deck falls back to a MapView if there are no views defined.